lib/eventboss/cli.rb

```
@@ -0,0 +1,150 @@
+      # Due to a changes introduced in Rails 6 regarding autoloading
+      # we need to determine how to perform eager_load
+      # @see https://weblog.rubyonrails.org/2019/2/22/zeitwerk-integration-in-rails-6-beta-2/
+      if ::Rails.try(:autoloaders).try(:zeitwerk_enabled?)
+        ::Zeitwerk::Loader.eager_load_all
+      else
+        ::Rails.application.eager_load!
+      end
+    rescue LoadError
+      Eventboss.logger.debug('Seems like not a Rails app')
+
+      if options[:require].nil?
+        Eventboss.logger.warn('Please use -r to load a custom app entrypoint')
+        exit(0)
+      else
+        Eventboss.logger.debug("Loading #{options[:require]}")
+        require File.expand_path(options[:require])
+      end
+    end
+
+    def parse_options(args)
+      option_parser(options).parse!(args)
+
+      options
+    end
+
+    def load_config_file
+      # check config file presence
+      if options[:config]
+        raise ArgumentError, "No such file #{options[:config]}" unless File.exist?(options[:config])
+      else
+        config_dir = if File.directory?(options[:require].to_s)
+                       File.join(options[:require], 'config')
+                     else
+                       File.join(DEFAULT_OPTIONS[:require], 'config')
+                     end
+
+        %w[eventboss.yml eventboss.yml.erb].each do |config_file|
+          path = File.join(config_dir, config_file)
+          options[:config] ||= path if File.exist?(path)
+        end
+      end
+
+      # parse config file options
+      if options[:config]
+        opts = parse_config(options[:config])
+
+        opts.each do |option_name, option|
+          if Eventboss::Configuration::OPTS_ALLOWED_IN_CONFIG_FILE.include?(option_name)
+            Eventboss.configuration.public_send("#{option_name}=", option)
+          else
+            Eventboss.logger.warn("Not supported option (#{option_name}) provided in config file.")
+            exit(1)
```

if not supported option is passed, should we fail hard or silently ignore it? 
it should either log error not warn, or shouldn't exit

---

Changed warn to error